src/actions.js

```diff
@@ -350,52 +366,81 @@ export const startWallet = (words, pin) => async (dispatch) => {
+      tokensBalance[token] = {
+        available: tokenBalance.unlocked,
+        locked: tokenBalance.locked,
+      };
+      const history = await wallet.getTxHistory({ token_id: token });
+      tokensHistory[token] = history.map((element) => mapTokenHistory(element, token));
+      /* eslint-enable no-await-in-loop */
+    }
+
+    dispatch(partiallyUpdateHistoryAndBalance({ tokensHistory, tokensBalance }));
+  };
+
   wallet.start({ pinCode: pin, password: pin }).then((serverInfo) => {
     walletUtil.storePasswordHash(pin);
     walletUtil.storeEncryptedWords(words, pin);
+
     dispatch(setServerInfo({ version: null, network: networkName }));
 
-    if (!useWalletService) {
-      wallet.on('new-tx', (tx) => {
-        fetchNewTxTokenBalance(wallet, tx).then((updatedBalanceMap) => {
-          if (updatedBalanceMap) {
-            dispatch(newTx(tx, updatedBalanceMap));
-          }
-        });
-      });
+    wallet.on('new-tx', (tx) => {
+      fetchNewTxTokenBalance(wallet, tx).then(async (updatedBalanceMap) => {
+        if (updatedBalanceMap) {
+          dispatch(newTx(tx, updatedBalanceMap));
 
-      wallet.on('update-tx', (tx) => {
-        fetchNewTxTokenBalance(wallet, tx).then((updatedBalanceMap) => {
-          if (updatedBalanceMap) {
-            dispatch(updateTx(tx, updatedBalanceMap));
+          if (useWalletService) {
+            handlePartialUpdate(updatedBalanceMap);
           }
-        });
+        }
       });
+    });
 
-      wallet.conn.on('best-block-update', (height) => {
-        fetchNewHTRBalance(wallet).then((data) => {
-          if (data) {
-            dispatch(updateHeight(height, data));
+    wallet.on('update-tx', (tx) => {
+      fetchNewTxTokenBalance(wallet, tx).then((updatedBalanceMap) => {
+        if (updatedBalanceMap) {
+          dispatch(updateTx(tx, updatedBalanceMap));
+
+          // Read comment on new-tx event
+          if (useWalletService) {
+            handlePartialUpdate(updatedBalanceMap);
           }
-        });
+        }
       });
+    });
 
-      wallet.conn.on('state', (state) => {
-        let isOnline;
-        if (state === Connection.CONNECTED) {
-          isOnline = true;
-        } else {
-          isOnline = false;
+    wallet.conn.on('best-block-update', (height) => {
+      fetchNewHTRBalance(wallet).then((data) => {
+        if (data) {
+          dispatch(updateHeight(height, data));
         }
-        dispatch(setIsOnline(isOnline));
       });
+    });
 
-      wallet.conn.on('wallet-load-partial-update', (data) => {
-        const transactions = Object.keys(data.historyTransactions).length;
-        const addresses = data.addressesFound;
-        dispatch(updateLoadedData({ transactions, addresses }));
-      });
-    }
+    wallet.conn.on('state', (state) => {
+      let isOnline;
+      if (state === Connection.CONNECTED) {
+        isOnline = true;
+      } else {
+        isOnline = false;
+      }
+      dispatch(setIsOnline(isOnline));
```

Comment:
`dispatch(setIsOnline(state == Connection.CONNECTED));` ? Would this work?

Reply:
It would! Thanks

Done in 4e4f7c7c57df45526b1e89847ca9befdce654b94